src/libstore/local-store.cc

```diff
@@ -281,6 +285,25 @@ void LocalStore::openDB(State & state, bool create)
             SQLITE_OPEN_READWRITE | (create ? SQLITE_OPEN_CREATE : 0), 0) != SQLITE_OK)
         throw Error(format("cannot open Nix database ‘%1%’") % dbPath);
 
+#ifdef __CYGWIN__
+    /* The cygwin version of sqlite3 has a patch which calls
+       SetDllDirectory("/usr/bin") on init. It was intended to fix extension
+       loading, which we don't use, and the effect of SetDllDirectory is
+       inherited by child processes, and causes libraries to be loaded from
+       /usr/bin instead of $PATH. This breaks quite a few things (e.g.
+       checkPhase on openssh), so we set it back to default behaviour. */
+    {
+        HMODULE kernel32 = GetModuleHandleW(L"kernel32.dll");
+        if (kernel32)
+        {
+            BOOL WINAPI (*setDllDirectory)(LPCWSTR) =
+                (BOOL WINAPI (*)(LPCWSTR))GetProcAddress(kernel32, "SetDllDirectoryW");
```

Comment:
I borrowed that from cygwin.  I presume it's for ancient versions of windows.  The API was added in XP/2003, so it was extreme overkill.  I've switched it over.

This will change build behaviour.  Is there some sort of global key I should increment to force rebuilds on cygwin?